main.ts

```diff
@@ -82,7 +82,13 @@ async function runSetup() {
         // Fetch the Guild using Guild ID and Channel using Channel ID
         const guild = await client.guilds.fetch(GUILD_ID);
         const channel = await guild.channels.fetch(CHANNEL_ID);
-
+        // On error, try to reconnect
+        socket.onerror = async (event) => {
+            console.log("WS connection error, sleeping, cleaning up and trying to reconnect again");
+            await sleep(1000);
+            cleanUp();
+            runSetup();
+        }
```

Comment:
curious, will this succeed if the internet is disconnected for a while? (like a minute)

Reply:
Yup, tried w/ simulating a dc by just pulling the LAN cable lol

Reply:
i laughed out loud